Hazel/Dtls/DtlsUnityConnection.cs

```diff
@@ -576,13 +594,21 @@ private bool ProcessHandshake(ref Record record, ByteSpan message)
                             continue;
                         }
 
-                        // Save the cookie
+                        // If the cookie differs, save it and restart the handshake
+                        if (this.nextEpoch.Cookie.Length == helloVerifyRequest.Cookie.Length
```

Comment:
I think we set the nextEpoch.Cookie to an empty one, but I wonder if there's a possible state where it is null? Also wondering if we have to worry about a helloVerifyRequest.Cookie being 0 length for whatever reason

Reply:
I don't think it's reasonable to ever expect a null Cookie. ByteSpans are structs so it itself can't be null and I would say if a null, yet non-zero-length internal array cropped up, it should be painstakingly fixed at the root cause. But a zero-length request cookie is interesting and I suppose a valid concern, so perhaps I'll wrap this Const method into a zero-length-tolerant variant.

Reply:
Actually... I guess if both cookies are zero length... I guess that's ok? Weird and hopefully not something a good server would do, but ... technically valid maybe?

Reply:
Noted about the details, indeed no null case then! And yeah... it might technically be valid, perhaps in some kind of testing scenario, but the concern might be some sort of falsely acting server? ie might be security for a client to know they are connecting to a legitimate entity, but I'm not sure - just thinking wide so far